pkg/webhook/admission/validator.go

```diff
@@ -0,0 +1,169 @@
+package admission
+
+import (
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+const (
+	// NicInterfaceCheckID defines an ID of a NIC interface model check
+	NicInterfaceCheckID = CheckID("nic.interface")
+	// NicOnBootID defines an ID of a NIC on_boot == fales check
+	NicOnBootID = CheckID("nic.on_boot")
+	// NicPluggedID defines an ID of a NIC plugged == false check
+	NicPluggedID = CheckID("nic.plugged")
+	// NicVNicPortMirroringID defines an ID of a vnic_profile.port_mirroring == true check
+	NicVNicPortMirroringID = CheckID("nic.vnic_profile.port_mirroring")
+	// NicVNicPassThroughID defines an ID of a vnic_profile.pass_through == 'enabled' check
+	NicVNicPassThroughID = CheckID("nic.vnic_profile.pass_through")
+	// NicVNicCustomPropertiesID defines an ID of a vnic_profile.custom_properties presence check
+	NicVNicCustomPropertiesID = CheckID("nic.vnic_profile.custom_properties")
+	// NicVNicNetworkFilterID defines an ID of a vnic_profile.networ_filter presence check
+	NicVNicNetworkFilterID = CheckID("nic.vnic_profile.network_filter")
+	// NicVNicQosID defines an ID of a vnic_profile.qos presence check
+	NicVNicQosID = CheckID("nic.vnic_profile.qos")
+)
+
+var validInterfaceDeviceModels = map[string]*struct{}{"e1000": nil, "rtl8139": nil, "virtio": nil}
+
+// CheckID identifies validation check for Virtual Machine Import
+type CheckID string
```

Comment:
How are we going to differentiate which `ValidationFailure` should be logged or we need to warn or block the import process? We could add validation level field which would allow us to interpret the slice with failures.

Reply:
I saw two options there: one was what you suggest, the other (chosen) was to separate the concerns of "action" assignment and detection of suspicious configuration elements. I decided to allow the client of the validation methods to decide how to treat each of the failures based on client-owned assignment of `CheckID` to action (i.e. map[CheckID]Action). Despite the other solution being simpler in implementation I feel that the 'level' does not belong to the `ValidationFailure`.  
Do you think the other approach would be better/clearer/cleaner?

Reply:
We know that actions when specific rules will change over time as features supported by Kubevirt will grow. We need to have this mapping somewhere and update it over time. My idea was to have the mapping together with rule definition and separate action implementation. My intention is simplicity so I would love to see what you have in mind in the code.

Reply:
+1 for simplicity. I'd add a category for each check as mentioned elsewhere (audit / warn / block) and fail the validation on 'block' violations.

Reply:
I was thinking about something like 
```
type action int

const (
	log   = 0
	warn  = 1
	block = 2
)

var checkToAction = map[CheckID]action{
	NicInterfaceCheckID:       block,
	NicOnBootID:               log,
	NicPluggedID:              warn,
	NicVNicPassThroughID:      block,
	NicVNicPortMirroringID:    warn,
	NicVNicCustomPropertiesID: warn,
	NicVNicNetworkFilterID:    warn,
	NicVNicQosID:              log,
}
```
in vm-import-validation.go
and then instead of getting the action from the `ValidationFailure` object, I would look it up in the  `checkToAction` whil processing the slice of failures.
I'm fine with having it in the `ValidationFailure`, though.

Reply:
The checkToAction map will be implemented in a different PR.